editoast/src/views/work_schedules.rs

```diff
@@ -275,6 +294,226 @@ async fn project_path(
+    responses(
+        (status = 200, body = WorkScheduleGroupCreateResponse, description = "The id of the created work schedule group"),
+    )
+)]
+async fn create_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    Json(WorkScheduleGroupCreateForm {
+        work_schedule_group_name,
+    }): Json<WorkScheduleGroupCreateForm>,
+) -> Result<Json<WorkScheduleGroupCreateResponse>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_name = work_schedule_group_name.unwrap_or(Uuid::new_v4().to_string());
+
+    // Create the work_schedule_group
+    let work_schedule_group = WorkScheduleGroup::changeset()
+        .name(group_name.clone())
+        .creation_date(Utc::now())
+        .create(conn)
+        .await;
+    let work_schedule_group = work_schedule_group.map_err(|e| map_diesel_error(e, group_name))?;
+    Ok(Json(WorkScheduleGroupCreateResponse {
+        work_schedule_group_id: work_schedule_group.id,
+    }))
+}
+
+#[utoipa::path(
+    delete, path = "",
+    tag = "work_schedules",
+    params(WorkScheduleGroupIdParam),
+    responses(
+        (status = 204, description = "The work schedule group has been deleted"),
+        (status = 404, description = "The work schedule group does not exist"),
+    )
+)]
+async fn delete_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    raw_group_id: Path<WorkScheduleGroupIdParam>,
+) -> Result<impl IntoResponse> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_id = raw_group_id.id;
+
+    // Check that the group exists
+    let work_schedule_group = WorkScheduleGroup::retrieve_or_fail(conn, group_id, || {
+        WorkScheduleError::WorkScheduleGroupNotFound { id: group_id }
+    })
+    .await?;
+
+    // List work schedules to be deleted
+    let selection_setting =
+        SelectionSettings::new().filter(move || WorkSchedule::WORK_SCHEDULE_GROUP_ID.eq(group_id));
+    let work_schedule_ids = WorkSchedule::list(conn, selection_setting)
+        .await?
+        .into_iter()
+        .map(|work_schedule| work_schedule.id);
+
+    // Delete them
+    use crate::models::DeleteBatch;
+    let conn = &mut db_pool.get().await?;
+    WorkSchedule::delete_batch(conn, work_schedule_ids).await?;
+
+    // Delete the group itself
+    WorkScheduleGroup::delete(&work_schedule_group, conn).await?;
```

Comment:
Fixed in f937a335fad70ebc83e5bc3a0992a73e553d2b67

Reply:
If I am not mistaken in this [comment](https://github.com/OpenRailAssociation/osrd/pull/9785#discussion_r1852372121) the delete_batch and the transaction might not be needed. What do you think @woshilapin ?

Reply:
Well if you only have to delete the `work_schedule_group` and the DB will do the rest, then yes, for sure it should be fine without a transaction. I'm resolving.

Reply:
Looking again at the code, I'm actually wondering why we want to `retrieve` first? Is it only for reporting an error in case the identifier doesn't exist? Can't we report that directly from a failure of the `delete`?

Reply:
Good point, the `delete` method wanted a group itself and not just an ID, but there seems to be other methods for that.

I haven't properly setup editoast on my laptop, I'll fix this tomorrow